tests/cli/test_remote.py

```diff
@@ -1,7 +1,731 @@
+        transforms = [some_transform]
+        remote = SlurmRemote(
+            mock_connection, alloc=alloc, transforms=transforms, persist=persist
+        )
+        persisted = remote.persist()
+
+        # NOTE: Feels dumb to do this. Not sure what I should be doing otherwise.
+        assert persisted.connection == remote.connection
+        assert persisted.alloc == remote.alloc
+        assert persisted.transforms == [
+            some_transform,
+            persisted.srun_transform_persist,
+        ]
+        assert persisted._persist is True
+
+    @dont_run_for_real
+    @disable_internet_access
+    def test_ensure_allocation_persist(self, mock_connection: Connection):
+        alloc = ["--time=00:01:00"]
+        transforms = [some_transform]
+        remote = SlurmRemote(
+            mock_connection, alloc=alloc, transforms=transforms, persist=True
+        )
+
+        # TODO: Not sure if this test has any use at this point..
+        remote.extract = Mock(
+            spec=remote.extract,
+            spec_set=True,
+            return_value=(
+                Mock(spec=invoke.runners.Runner, spec_set=True),
+                {"node_name": "bob", "jobid": "1234"},
+            ),
+        )
+
+        results, runner = remote.ensure_allocation()
+
+        remote.extract.assert_called_once_with(
+            "echo @@@ $(hostname) @@@ && sleep 1000d",
+            patterns={
+                "node_name": "@@@ ([^ ]+) @@@",
+                "jobid": "Submitted batch job ([0-9]+)",
+            },
+            hide=True,
+        )
+        assert results == {"node_name": "bob", "jobid": "1234"}
+        # raise NotImplementedError("TODO: Imporant and potentially complicated test")
+
+    @dont_run_for_real
+    @disable_internet_access
+    def test_ensure_allocation_without_persist(self, mock_connection: Connection):
+        alloc = ["--time=00:01:00"]
+        transforms = [some_transform]
+        remote = SlurmRemote(
+            mock_connection, alloc=alloc, transforms=transforms, persist=False
+        )
+
+        def write_stuff(
+            command: str,
+            asynchronous: bool,
+            hide: bool,
+            warn: bool,
+            pty: bool,
+            out_stream: QueueIO,
+        ):
+            assert command == f"bash -c 'salloc {shjoin(alloc)}'"
+            out_stream.write("salloc: Nodes bob-123 are ready for job")
+            return unittest.mock.DEFAULT
+
+        mock_connection.run.side_effect = write_stuff
+        results, runner = remote.ensure_allocation()
+
+        mock_connection.run.assert_called_once_with(
+            f"bash -c 'salloc {shjoin(alloc)}'",
+            hide=False,
+            asynchronous=True,
+            out_stream=unittest.mock.ANY,
+            pty=True,
+            warn=False,
+        )
+        assert results == {"node_name": "bob-123"}
+        # raise NotImplementedError("TODO: Imporant and potentially complicated test")
```

Comment:
Same, I think we could mock and use a `FileRegressionFixture` or something like that

Reply:
I don't see how that would work, perhaps we can chat about this in person